src/firstvoices/backend/tests/test_dictionary_models/test_parts_of_speech.py

```diff
@@ -0,0 +1,70 @@
+import json
+
+import pytest
+from django.core.management import call_command
+from django.shortcuts import get_list_or_404
+from django.utils import timezone
+from rest_framework.reverse import reverse
+from rest_framework.test import APIClient
+
+from firstvoices.backend.models.constants import AppRole
+from firstvoices.backend.models.part_of_speech import PartOfSpeech
+from firstvoices.backend.tests.factories import get_app_admin
+
+
+# Fixtures
+@pytest.fixture(scope="session")
+def parts_of_speech_db_setup(django_db_setup, django_db_blocker):
+    with django_db_blocker.unblock():
+        call_command("loaddata", "partsOfSpeech_initial.json")
+
+
+class TestPartsOfSpeechModel:
+    """Tests for parts-of-speech model."""
+
+    def test_save_operation(self, db, parts_of_speech_db_setup):
+        # Testing save operation for the model
+        p1 = PartOfSpeech(
+            title="part_of_speech_1",
+            created=timezone.now(),
+            last_modified=timezone.now(),
+        )
+        p1.save()
+        p2 = PartOfSpeech(
+            title="part_of_speech_2",
+            parent=p1,
+            created=timezone.now(),
+            last_modified=timezone.now(),
+        )
+        p2.save()
+
+        p1_fetch = PartOfSpeech.objects.get(title="part_of_speech_1")
+        assert p1_fetch.title == "part_of_speech_1", "Title does not match"
+        p2_fetch = PartOfSpeech.objects.get(title="part_of_speech_2")
+        assert p2_fetch.title == "part_of_speech_2", "Title does not match"
+        assert (
+            p2_fetch.parent.title == "part_of_speech_1"
+        ), "Parent's title does not match"
+
+    def test_fixture_with_model(self, db, parts_of_speech_db_setup):
+        # Testing that the fixture works, and we have at least 1 object in the database
+        pos_count = get_list_or_404(PartOfSpeech)
+        assert len(pos_count) != 0
+
+
+class TestPartsOfSpeechView:
+    """Tests for parts-of-speech views."""
```

Comment:
Are there more tests coming for the API?

Reply:
I don't have any specific ideas as of now, but would like to add more tests to make it more well tested.

Reply:
It might be worth testing that each user type, (superadmin, staff, language admin, editor, assistant, member, anonymous), can see the parts of speech, duplicate titles aren't possible, and testing that grandparent creation is not possible.

Reply:
Thanks @NolanVH for the ideas, these can be great additions. Since there is not Create/Update/Delete as of now for this view, so I didn't add some of the tests you have mentioned as of now. I will add the grandparent creation check and a test to check if diff. user roles can see the parts of speech though which seems like good additions.

Reply:
It would also be good to test that the data format is as expected, since there aren't separate tests for the serializer. At a minimum, the number of items in the returned list and the property names for one of them.